File: app/src/main/java/com/example/breezil/favnews/di/module/MainActivityModule.java
package com.example.breezil.favnews.di.module;

import com.example.breezil.favnews.ui.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class MainActivityModule {
    @ContributesAndroidInjector(modules = MainFragmentsModule.class)
    abstract MainActivity contributeMainActivity();
}

File: app/src/main/java/com/example/breezil/fave/di/module/MainActivityModule.kt
package com.example.breezil.fave.di.module

import com.example.breezil.fave.ui.main.MainActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class MainActivityModule {
    @ContributesAndroidInjector(modules = [MainFragmentsModule::class])
    internal abstract fun contributeMainActivity(): MainActivity
}